### DISCUSSION TITLE
Preventing infinite release cycle with circular dependency and depsUpgrade: true

### DISCUSSION THREAD

**Discussion by vladcosorg** [projen/projen#2722] [Q&A] [answered]
So I got 2 projen-based packages that list each other as dependencies in package.json. Both have depsUpgrade enabled with auto-approve and auto-merge. Obviously now we have a problem that when the scheduled `upgrade` workflow runs, they bump each other versions continuously. 

I am aware that I can break the cycle by excluding the dependency from upgrading on one end or another.

Is there another solution that I'm overlooking that would still let me keep all the deps upgraded automatically? For example bumping the version ONLY when there are other upgrades except the cyclic one?

Any suggestions? Thanks!

**Comment by mrgrain** [comment, marked as answer]
You can add a condition to the release task that does any custom check you want it to do.

```ts
project
  .tryFindObjectFile('.projen/tasks.json')
  .patch(JsonPatch.add('/tasks/release/condition', 'node ./scripts/should-release.js'));
```

**Comment by vladcosorg** [comment]
Awesome solution! Thanks!
